api/v1alpha1/shared_types.go

```diff
@@ -209,6 +209,8 @@ type KubernetesPodSpec struct {
 
 // KubernetesContainerSpec defines the desired state of the Kubernetes container resource.
 // +kubebuilder:validation:XValidation:rule="!has(self.image) || !has(self.imageRepository)",message="Either image or imageRepository can be set."
+// +kubebuilder:validation:XValidation:rule="has(self.image) ? self.image.matches('^[^:]+:[^:]+$') : true",message="Image must include a tag (e.g., 'image:tag')."
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="self.image.matches('^[a-zA-Z0-9._/-]+:[a-zA-Z0-9._-]+$') : true",message="Image must include a tag and allowed characters only (e.g., 'image:tag')."
```
Added some suggestions to make regex stricter. Also can we remove `has` checks, since kube validations apply on non-null values only?

Reply:
@sudiptob2 can we do something like this?
https://github.com/envoyproxy/gateway/blob/ddc9a5c169877f13e0ce6b0ece3fe820df83ee25/api/v1alpha1/shared_types.go#L339

Reply:
updated